src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -1438,6 +1439,34 @@ internal string[] ShortTimes
             }
         }
 
+        private string[]? GetTimeFormatsCore()
```

Comment:
>GetTimeFormatsCore [](start = 26, length = 18)

should we have 2 copies of such methods one for Windows and one for Linux? just to avoid the ShouldUseUserOverrideNlsData check on Linux and to ensure running the rest of the code on Windows only. 

otherwise we can keep this way but adding assert after this check to ensure we are running on Windows.

Reply:
Yeah we can do that.